Intersect.Client/Interface/Game/DescriptionWindows/SpellDescriptionWindow.cs

```diff
@@ -0,0 +1,406 @@
+                else if (mSpell.Combat.StatDiff[i] > 0)
+                {
+                    data = new Tuple<string, string>(Strings.SpellDesc.StatCounts[i], mSpell.Combat.StatDiff[i].ToString());
+                }
+                else if (mSpell.Combat.PercentageStatDiff[i] > 0)
+                {
+                    data = new Tuple<string, string>(Strings.SpellDesc.StatCounts[i], Strings.ItemDescription.Percentage.ToString(mSpell.Combat.PercentageStatDiff[i]));
+                }
+
+                // Make sure we only add a blank row the first time we add a stat row.
+                if (data != null)
+                {
+                    if (!blankAdded)
+                    {
+                        mDetailRows.AddKeyValueRow(string.Empty, string.Empty);
+                        mDetailRows.AddKeyValueRow(Strings.SpellDesc.StatBuff, string.Empty);
+                        blankAdded = true;
+                    }
+
+                    mDetailRows.AddKeyValueRow(data.Item1, data.Item2);
+                    showDuration = true;
+                }
+            }
+
+            // Handle HoT and DoT displays.
+            if (mSpell.Combat.HoTDoT)
+            {
+                showDuration = true;
+                mDetailRows.AddKeyValueRow(string.Empty, string.Empty);
+                if (isHeal)
+                {
+                    mDetailRows.AddKeyValueRow(Strings.SpellDesc.HoT, string.Empty);
+                } 
+                else if (isDamage)
+                {
+                    mDetailRows.AddKeyValueRow(Strings.SpellDesc.DoT, string.Empty);
+                }
+                mDetailRows.AddKeyValueRow(Strings.SpellDesc.Tick, Strings.SpellDesc.Seconds.ToString(mSpell.Combat.HotDotInterval / 1000f));
+            }
+
+            // Handle effect display.
+            if (mSpell.Combat.Effect != StatusTypes.None)
+            {
+                showDuration = true;
+                mDetailRows.AddKeyValueRow(string.Empty, string.Empty);
+                mDetailRows.AddKeyValueRow(Strings.SpellDesc.Effect, Strings.SpellDesc.Effects[(int) mSpell.Combat.Effect]);
+            }
+
+            // Show Stat Buff / Effect / HoT / DoT duration.
+            if (showDuration)
+            {
+                mDetailRows.AddKeyValueRow(Strings.SpellDesc.Duration, Strings.SpellDesc.Seconds.ToString(mSpell.Combat.Duration / 1000f));
+            }
+
+            // Resize and position the container.
+            mDetailRows.SizeToChildren(true, true);
+        }
+
+        protected void SetupDashInfo()
+        {
+            // Add a divider.
+            mDivider3 = AddDivider();
+
+            // Add a row component.
+            mDetailRows = AddRowContainer();
+
+            // Dash Distance Information.
+            mDetailRows.AddKeyValueRow(Strings.SpellDesc.Distance, Strings.SpellDesc.Tiles.ToString(mSpell.Combat.CastRange));
+
+            // Ignore map blocks?
+            if (mSpell.Dash.IgnoreMapBlocks)
```

Comment:
Same as my last suggestion. Maybe you can use a Dictionary<bool, string> here too.